### ISSUE TITLE
"Could not load file or assembly 'System.Runtime'" when running benchmarks with local runtime build on Ubuntu 20.04

### ISSUE THREAD

**Issue by alexcovington** [dotnet/performance#1422]
I'm trying to benchmark a local build of CoreCLR on Ubuntu 20.04 but am getting an error whenever I try to point to the `corerun` binary.

When I run the benchmarks normally with the system-installed runtime with the following command:

```
dotnet run -c Release -f netcoreapp5.0
```

everything runs normally as expected and any benchmark I select from the interactive menu finishes running.

However, when I run and pass the path to the `corerun` binary with this command:

```
dotnet run -c Release -f netcoreapp5.0 --corePath /home/acovingt/projects/runtime/artifacts/bin/coreclr/Linux.x64.Release/corerun
```

I am prompted to select a specific benchmark but will get an error in the output regardless of which benchmark I select:

```
// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 3 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet restore /p:DebugType=portable /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 in /home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/87faeee9-e308-42d9-ac27-31409fd6ebee
// command took 1.51s and exited with 0
// start dotnet build -c Release /p:DebugType=portable --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 in /home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/87faeee9-e308-42d9-ac27-31409fd6ebee
// command took 19.31s and exited with 1
// start dotnet build -c Release /p:DebugType=portable --no-restore --no-dependencies /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 in /home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/87faeee9-e308-42d9-ac27-31409fd6ebee
// command took 1.87s and exited with 0
// start dotnet publish -c Release /p:DebugType=portable --no-build --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 in /home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/87faeee9-e308-42d9-ac27-31409fd6ebee
// command took 1.03s and exited with 0
// ***** Done, took 00:00:24 (24.23 sec)   *****
// Found 3 benchmarks:
//   BilinearTest.Interpol_Scalar: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
//   BilinearTest.Interpol_Vector: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
//   BilinearTest.Interpol_AVX: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)

// **************************
// Benchmark: BilinearTest.Interpol_Scalar: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
// *** Execute ***
// Launch: 1 / 1
// Execute: /home/acovingt/projects/runtime/artifacts/bin/coreclr/b379c2e1-9ff1-49c1-9f5c-4351aa778c86/corerun "87faeee9-e308-42d9-ac27-31409fd6ebee.dll" --benchmarkName "BilinearTest.Interpol_Scalar" --job "PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1" --benchmarkId 0 in /home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/87faeee9-e308-42d9-ac27-31409fd6ebee/bin/Release/netcoreapp5.0/publish
Failed to set up high priority. Make sure you have the right permissions. Message: Permission denied
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.

File name: 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
ExitCode != 0
// Benchmark Process 426750 has exited with code 134
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// **************************
// Benchmark: BilinearTest.Interpol_Vector: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
// *** Execute ***
// Launch: 1 / 1
// Execute: /home/acovingt/projects/runtime/artifacts/bin/coreclr/b379c2e1-9ff1-49c1-9f5c-4351aa778c86/corerun "87faeee9-e308-42d9-ac27-31409fd6ebee.dll" --benchmarkName "BilinearTest.Interpol_Vector" --job "PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1" --benchmarkId 1 in /home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/87faeee9-e308-42d9-ac27-31409fd6ebee/bin/Release/netcoreapp5.0/publish
Failed to set up high priority. Make sure you have the right permissions. Message: Permission denied
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.

File name: 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
ExitCode != 0
// Benchmark Process 426760 has exited with code 134
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// **************************
// Benchmark: BilinearTest.Interpol_AVX: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
// *** Execute ***
// Launch: 1 / 1
// Execute: /home/acovingt/projects/runtime/artifacts/bin/coreclr/b379c2e1-9ff1-49c1-9f5c-4351aa778c86/corerun "87faeee9-e308-42d9-ac27-31409fd6ebee.dll" --benchmarkName "BilinearTest.Interpol_AVX" --job "PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1" --benchmarkId 2 in /home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/87faeee9-e308-42d9-ac27-31409fd6ebee/bin/Release/netcoreapp5.0/publish
Failed to set up high priority. Make sure you have the right permissions. Message: Permission denied
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.

File name: 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
ExitCode != 0
// Benchmark Process 426769 has exited with code 134
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// ***** BenchmarkRunner: Finish  *****

// * Export *
  home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/BenchmarkDotNet.Artifacts/results/BilinearTest-report.csv
  home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/BenchmarkDotNet.Artifacts/results/BilinearTest-report-github.md
  home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/BenchmarkDotNet.Artifacts/results/BilinearTest-report.html
  home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/BenchmarkDotNet.Artifacts/results/BilinearTest-report-full.json
  home/acovingt/projects/performance/artifacts/bin/MicroBenchmarks/Release/netcoreapp5.0/BenchmarkDotNet.Artifacts/results/BilinearTest-perf-lab-report.json

// * Detailed results *
BilinearTest.Interpol_Scalar: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
Runtime = ; GC = 
There are not any results runs

BilinearTest.Interpol_Vector: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
Runtime = ; GC = 
There are not any results runs

BilinearTest.Interpol_AVX: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
Runtime = ; GC = 
There are not any results runs

// * Summary *

BenchmarkDotNet=v0.12.1, OS=ubuntu 20.04
AMD Ryzen 5 3600, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=5.0.100-rc.1.20378.7
  [Host] : .NET Core 5.0.0 (CoreCLR 5.0.20.37113, CoreFX 5.0.20.37113), X64 RyuJIT

PowerPlanMode=00000000-0000-0000-0000-000000000000  Arguments=/p:DebugType=portable  Toolchain=CoreRun  
IterationTime=250.0000 ms  MaxIterationCount=20  MinIterationCount=15  
WarmupCount=1  

|          Method | Mean | Error | Median | Min | Max |
|---------------- |-----:|------:|-------:|----:|----:|
| Interpol_Scalar |   NA |    NA |     NA |  NA |  NA |
| Interpol_Vector |   NA |    NA |     NA |  NA |  NA |
|    Interpol_AVX |   NA |    NA |     NA |  NA |  NA |

Benchmarks with issues:
  BilinearTest.Interpol_Scalar: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
  BilinearTest.Interpol_Vector: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)
  BilinearTest.Interpol_AVX: Job-GKFAPL(PowerPlanMode=00000000-0000-0000-0000-000000000000, Arguments=/p:DebugType=portable, Toolchain=CoreRun, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1)

// * Legends *
  Mean   : Arithmetic mean of all measurements
  Error  : Half of 99.9% confidence interval
  Median : Value separating the higher half of all measurements (50th percentile)
  Min    : Minimum
  Max    : Maximum
  1 ns   : 1 Nanosecond (0.000000001 sec)

// * Diagnostic Output - MemoryDiagnoser *


// ***** BenchmarkRunner: End *****
// ** Remained 0 benchmark(s) to run **
Run time: 00:00:02 (2.11 sec), executed benchmarks: 3

Global total time: 00:00:26 (26.34 sec), executed benchmarks: 3
// * Artifacts cleanup *
```

The [documentation](https://github.com/dotnet/performance/blob/master/docs/benchmarking-workflow-dotnet-runtime.md#dotnet-runtime-prerequisites) doesn't use this same path when getting CoreRun.exe on Windows, so I could be grabbing the wrong binary. However, `corerun` doesn't exist in the `artifacts/bin/testhost` directory anywhere.

`dotnet --info` output for more troubleshooting info:

```
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.1.20378.7
 Commit:    de16eb81b7

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /home/acovingt/.local/dotnet/sdk/5.0.100-rc.1.20378.7/

Host (useful for support):
  Version: 5.0.0-rc.1.20371.13
  Commit:  d8cf13e0ba

.NET SDKs installed:
  5.0.100-rc.1.20378.7 [/home/acovingt/.local/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.0-rc.1.20377.7 [/home/acovingt/.local/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.0-rc.1.20371.13 [/home/acovingt/.local/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

Any help would be appreciated!

**Comment by danmosemsft** [comment]
How did you build coreclr? It should have copied corerun into testhost. You need to run it there so that it can find everything else.

Meantime, it might work to simply manually copy it there and run it from there.

**Comment by alexcovington** [comment]
@danmosemsft, I built CoreCLR with the following:

```
cd ~/projects/runtime # This is where the runtime repository is cloned
git clean -xdf
git pull origin master # git commands are part of my script, just to make sure I'm building the latest off of master
./build.sh -c Release -lc Release -rc Release -f netcoreapp5.0 --arch x64 --os Linux
```

This doesn't seem to produce a `corerun` in the `testhost` directory. The build also is not reporting any errors.

I will try copying `corerun` into the `testhost` directory and mimic the Windows directory structure and will update the issue with results.

**Comment by alexcovington** [comment]
Copying `corerun` seems to have fixed it.

**Comment by danmosemsft** [comment]
@jkoritzinsky should `./build.sh -c Release -lc Release -rc Release -f netcoreapp5.0 --arch x64 --os Linux` have copied corerun to the testhost directory? If not, what command would do that?

**Comment by jkoritzinsky** [comment]
I'm not sure. That's part of the libraries test infrastructure. @ViktorHofer would know better than I.

**Comment by danmosemsft** [comment]
duplicate: https://github.com/dotnet/runtime/issues/40215

**Comment by danmosemsft** [comment]
duplicate of https://github.com/dotnet/runtime/issues/40215

**Comment by danmosemsft** [comment]
Well, I guess Github duplicates don't work across repos...

**Comment by alexcovington** [comment]
@danmosemsft Strange, I appreciate the link to the other issue. I'll follow that one for a fix. Thanks for your help!
